### PR TITLE
Resample images to desired resolution

### LINKED ISSUE #91: Add image resampling resolution in config file
We are now testing the impact of image resampling on results. It would be nice to have an image resampling resolution parameter in the `config_script.yml` file.

### PR DESCRIPTION
**Pull request by PaulBautin**
It was noticed in issue #83 that the CSA of rescaled t1w images was often overestimated. This could be caused by an increasing partial volume effect with tissue outside of the CSF. However, overestimation is less visible in t2w images, this is possibly due to better image resolution (0.8x0.8x0.8mm instead of 1x1x1mm). By resampling the T2w data to 1mm at the very beginning, it would be possible to see if difference between t1w and t2w results is caused by difference of native resolution.

DONE:
- Resample t2w images

FIX #91

### CODE REVIEW

File: config_sct_run_batch.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # config file for sct_run_batch
 path_data: /scratch/pabaua/data-multi-subject-p
-path_output: /scratch/pabaua/results_csa_t2_2
+path_output: /scratch/pabaua/results_csa_t2_3
```
Comment (jcohenadad) [inline config_sct_run_batch.yml:3]: why these changes? they are local (ie specific to your config)

Comment (PaulBautin) [reply config_sct_run_batch.yml:3]: It is to keep track of the configs i used when i run jobs on compute canada. I will not merge this into master. In the future i will fork the repo for these kinds of PRs.

File: process_data.sh [hunk 2]
```diff
@@ -82,10 +82,11 @@ segment_if_does_not_exist(){
   local qc=$3
   # Update global variable with segmentation file name
   FILESEG="${file}_seg"
-  FILESEGMANUAL="${path_derivatives}/${FILESEG}-manual.nii.gz"
+  FILESEGMANUAL="${path_derivatives}/${FILESEG}-manual"
   if [ -e $FILESEGMANUAL ]; then
     echo "Found! Using manual segmentation."
-    rsync -avzh $FILESEGMANUAL ${FILESEG}.nii.gz
+    sct_resample -i ${FILESEGMANUAL}.nii.gz -mm $interp -x nn -o ${FILESEGMANUAL}_r.nii.gz
+    rsync -avzh ${FILESEGMANUAL}.nii.gz ${FILESEG}.nii.gz
```
Comment (jcohenadad) [inline process_data.sh:89]: problem here, the resampled `${FILESEGMANUAL}_r.nii.gz` should be copied to `${FILESEG}.nii.gz`

Comment (PaulBautin) [reply process_data.sh]: Suggestion was implemented in commit: https://github.com/sct-pipeline/csa-atrophy/pull/85/commits/482a70471cab25bc344959e6343175b1fdbc5066

File: process_data.sh [hunk 3]
```diff
@@ -113,7 +114,7 @@ cd anat
 # Reorient to RPI and resample file
 if [ $contrast == "t2" ]; then
   contrast_str="T2w"
-  interp="0.8x0.8x0.8"
+  interp="1x1x1"
```
Comment (jcohenadad) [inline process_data.sh:117]: i was suggesting to resample to 1mm iso only for "checking" the effect of this resampling, not for the "stable" release of the pipeline

Comment (PaulBautin) [reply process_data.sh:117]: Sorry for the confusion, i was thinking of re-running this PR on Compute Canada to put results in the article's supplementary material (It could partly explain differences observed between t1 and t2 results). @jcohenadad, what are your thoughts on that?

Comment (jcohenadad) [reply process_data.sh:117]: sounds good. IN fact, if we explore the effect of resampling, we probably want to also try other resamplings, e.g. 1.2mm iso

**Comment by PaulBautin** [comment]
I finally decided to implement issue #91 in this PR with commit https://github.com/sct-pipeline/csa-atrophy/pull/85/commits/6104cf5d9dd5b31b779ae55439d9ec5cb03861c7. @jcohenadad, i have tested this PR on a couple subjects and it works well. Could you review before merge?

File: config_script.yml [hunk 4]
```diff
@@ -14,8 +14,12 @@ rescaling:
   - 0.95
   - 0.94
   - 0.93
+
 # Contrasts on which to perform the analysis. Available values are: 't1', 't2'
 contrast: "t2"
+# image resolution (generally T1w images have resolution 1mm isotropic and T2w images 0.8mm isotropic)
```
Comment (jcohenadad) [inline config_script.yml:20]:
```suggestion
# Resampling resolution in mm. Native resolutions for the spine-generic project are 1mm isotropic for T1w images and 0.8mm isotropic for T2w images. 
```

**Review by jcohenadad — APPROVED** [review]

**Comment by PaulBautin** [comment]
Last commits allow user to resample images to desired resolution. This new parameter must be configured in config_script.yml before running program.
